core/cloudflow-operator/src/main/scala/cloudflow/operator/CloudflowApplication.scala

```diff
@@ -122,30 +122,75 @@ object CloudflowApplication {
   }
 
   object Status {
+    val Unknown          = "Unknown"
+    val Running          = "Running"
+    val Pending          = "Pending"
+    val CrashLoopBackOff = "CrashLoopBackOff"
+
     def apply(
-        app: CloudflowApplication.CR,
-        streamletStatuses: Vector[StreamletStatus] = Vector()
-    ): Status =
+        spec: CloudflowApplication.Spec
+    ): Status = {
+      val streamletStatuses = createStreamletStatuses(spec)
       Status(
-        app.spec.appId,
-        app.spec.appVersion,
-        streamletStatuses
+        spec.appId,
+        spec.appVersion,
+        streamletStatuses,
+        Unknown
       )
+    }
+
+    def createStreamletStatuses(spec: CloudflowApplication.Spec) = {
+      // TODO this code needs to be improved.
+      // There are some plans to make replicas mandatory
+      // and to indicate extraPods required in the deployment to prevent this issue
+      import cloudflow.operator.runner._
+      spec.deployments.map { deployment =>
+        val expectedPodCount = deployment.runtime match {
+          case AkkaRunner.runtime  ⇒ deployment.replicas.getOrElse(AkkaRunner.DefaultReplicas)
+          case SparkRunner.runtime ⇒ deployment.replicas.getOrElse(SparkRunner.DefaultNrOfExecutorInstances) + 1
+          case FlinkRunner.runtime ⇒ deployment.replicas.getOrElse(FlinkRunner.DefaultReplicas) + 1
+        }
+        StreamletStatus(
+          deployment.streamletName,
+          expectedPodCount
+        )
+      }
+    }
+
+    def calcAppStatus(streamletStatuses: Vector[StreamletStatus]): String =
+      if (streamletStatuses.forall { streamletStatus =>
+            streamletStatus.podStatuses.size == streamletStatus.expectedPodCount &&
+            streamletStatus.podStatuses.forall(_.isReady)
+          }) {
+        Status.Running
+      } else if (streamletStatuses.flatMap(_.podStatuses).exists(_.status == PodStatus.CrashLoopBackOff)) {
+        Status.CrashLoopBackOff
+      } else {
+        Status.Pending
+      }
   }
 
-  case class Status(
-      appId: String,
-      appVersion: String,
-      streamletStatuses: Vector[StreamletStatus]
-  ) {
+  // the status is created with the expected number of streamlet statuses, derived from the CloudflowApplication.Spec, see companion
+  case class Status private (appId: String, appVersion: String, streamletStatuses: Vector[StreamletStatus], appStatus: String) {
```

Comment:
enum (`case object`) for `appStatus` ?

Reply:
complicates the json format, and it is just the result that must be shown.